src/EditorFeatures/Core.Wpf/InlineRename/AbstractEditorInlineRenameService.SymbolRenameInfo.cs

```diff
@@ -241,6 +238,30 @@ public bool TryOnAfterGlobalSymbolRenamed(Workspace workspace, IEnumerable<Docum
                 return _refactorNotifyServices.TryOnAfterGlobalSymbolRenamed(workspace, changedDocumentIDs, RenameSymbol,
                     this.GetFinalSymbolName(replacementText), throwOnFailure: false);
             }
+
+            public InlineRenameFileRenameInfo GetFileRenameInfo()
+            {
+                var kindSupportsRename = RenameSymbol.Kind == SymbolKind.NamedType;
+                var allowRename = kindSupportsRename 
+                    && (RenameSymbol.Locations.Length == 1)
+                    && (OriginalNameMatches(_document, RenameSymbol.Name) || ContainsOnlyOneType(_document));
+
+                return new InlineRenameFileRenameInfo(kindSupportsRename, allowRename);
+
+                // Local Functions
+
+                static bool OriginalNameMatches(Document document, string name)
+                {
+                    return document.Name.Substring(0, document.Name.Length - 3) == name;
```

Comment:
what is the `-3` for?  removing the extension?  if so def don't do this.  There are utilities that will do the right thing for you.

Reply:
Yup, that was me forgetting that `Path` stuff exists. I meant to replace with something better before I pushed, and then forgot :)